Sources/RedCat/Service.swift

```diff
@@ -69,5 +68,29 @@ open class DetailService<State, Detail : Equatable> : Service<State> {
     open func onUpdate(newValue: Detail, store: Store<State>, environment: Dependencies) {
         
     }
-    
+}
+
+open class MapService<Base, State>: Service<State> {
```

Comment:
I used it for ComposedStore, but then did it differently, so perhaps MapService is unnecessary
In general, it would be convenient to make services not tied to specific State, like ErasedReducers

Reply:
The problem here is that many services do need to know *some* state, so I can't just make services preemptively agnostic about state. If you service doesn't need to know anything about the state (e.g. because you only look at actions), you can just make the service generic over the state. If you need to focus on a particular part of the global state, you can use a ``DetailService``` (or write your own version of that).